### PR TITLE
Add status field to Vault credentials and create private view

### PR DESCRIPTION
**Pull request by mgaffney**
This PR:
- creates `credential_vault_credential_status_enm`, a new enumeration table for the status of vault credentials
- adds a `status` column to the `credential_vault_credential` table with a foreign key relationship to the new `credential_vault_credential_status_enm` table
- creates `credential_vault_credential_private`, a new view in the database
- updates the protobuf and struct for the new status field
- updates the bits to save the new status field

### CODE REVIEW

File: internal/credential/vault/credential.go [hunk 1]
```diff
@@ -10,7 +10,37 @@ import (
...
+	// terminal status. It does not transition to ExpiredCredential.
+	RevokedCredential CredentialStatus = "revoked"
+
+	// ExpiredCredential represents a credential that expired. This is a terminal
+	// status. It does not transition to RevokedCredential.
+	ExpiredCredential CredentialStatus = "expired"
+)
+
+const (
+	// TODO(mgaffney) 06/2021: credentials with an externalIdSentinel value
```
Comment (louisruch) [inline internal/credential/vault/credential.go:36]: Not following this TODO, it seems you are handling this below

File: internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql [hunk 2]
```diff
@@ -428,4 +458,47 @@ begin;
...
+            cert.certificate_key_hmac    as client_cert_key_hmac,
+            cert.key_id                  as client_key_id
+       from credential_vault_credential credential
+       join credential_vault_token token
+         on credential.token_hmac = token.token_hmac
+       join credential_vault_store store
+         on token.store_id = store.public_id
+  left join credential_vault_client_certificate cert
+         on store.public_id = cert.store_id
+      where credential.expiration_time != 'infinity'::date;
```
Comment (louisruch) [inline internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql:498]: Interesting...is it valid to have `is_renewable = true` and an `expiration` of `infinity`?  I was going down the route of `where credential.is_renewable = true` so I am curious about how each of this fields interact with each other.

**Review by louisruch — APPROVED** [review]
looks great ty

File: internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql [hunk 3]
```diff
@@ -428,4 +458,47 @@ begin;
...
+            token.last_renewal_time      as token_last_renewal_time,
+            token.expiration_time        as token_expiration_time,
+            token.key_id                 as token_key_id,
+            token.status                 as token_status,
+            store.vault_address          as vault_address,
+            store.namespace              as namespace,
+            store.ca_cert                as ca_cert,
+            store.tls_server_name        as tls_server_name,
+            store.tls_skip_verify        as tls_skip_verify,
+            cert.certificate             as client_cert,
```
Comment (louisruch) [inline internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql:487]: I need the scope_id to create the database wrapper to decrypt: `store.scope_id as scope_id,`

File: internal/credential/vault/credential.go (hunk 1 above)
Comment (mgaffney) [reply internal/credential/vault/credential.go]: Fixed. Ty!

File: internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql (hunk 3 above)
Comment (mgaffney) [reply internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql:488]: Added

File: internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql (hunk 2 above)
Comment (mgaffney) [reply internal/db/schema/migrations/postgres/10/03_vault_credential.up.sql:499]: That is up to Vault. If a credential `is_renewable = true` and has an expiration of `infinity` why renew it?

**Pull request merged**